provider/datagen/src/driver.rs

```diff
@@ -334,74 +354,110 @@ impl DatagenDriver {
+                                            iter.get()
+                                        );
+                                        return Ok(());
+                                    } else {
+                                        // Not a match: we must include this
+                                        break;
+                                    }
                                 }
                             }
-                        }
-                        // Did not find a match: export this payload
-                        sink.put_payload(key, locale, payload).map_err(|e| {
-                            e.with_req(
-                                key,
-                                DataRequest {
-                                    locale,
-                                    metadata: Default::default(),
-                                },
-                            )
-                        })
-                    })?
-                }
-                FallbackMode::Hybrid | FallbackMode::Preresolved => {
-                    locales_to_export.into_par_iter().try_for_each(|locale| {
-                        if let Some(payload) = load_with_fallback(key, &locale) {
-                            sink.put_payload(key, &locale, &payload?)
-                        } else {
-                            Ok(())
-                        }
-                        .map_err(|e| {
-                            e.with_req(
-                                key,
-                                DataRequest {
-                                    locale: &locale,
-                                    metadata: Default::default(),
-                                },
-                            )
+                            // Did not find a match: export this payload
+                            sink.put_payload(key, locale, payload).map_err(|e| {
+                                e.with_req(
+                                    key,
+                                    DataRequest {
+                                        locale,
+                                        metadata: Default::default(),
+                                    },
+                                )
+                            })
+                        })?;
+
+                    // Slowest locale calculation:
+                    payloads
+                        .iter()
+                        .map(|(locale, (_payload, duration))| {
+                            (*duration, locale.write_to_string().into_owned())
                         })
-                    })?
+                        .max()
                 }
+                FallbackMode::Hybrid | FallbackMode::Preresolved => locales_to_export
+                    .into_par_iter()
+                    .map(|locale| {
+                        let instant2 = Instant::now();
+                        load_with_fallback(key, &locale)
```

Comment:
Do double-check that the old and new behaviors are the same. I switched things around to avoid duplicating the duration logic.